src/prefect/tasks.py

```diff
@@ -422,6 +422,11 @@ def __init__(
 
         self.task_key: str = _generate_task_key(self.fn)
 
+        # determine cache and result configuration
+        settings = get_current_settings()
+        if settings.tasks.default_no_cache and cache_policy is NotSet:
+            cache_policy = NO_CACHE
```

Comment:
There's a case down on line 466 that can set the cache policy to `None` after this. Is that something that we still want?

Reply:
wow good catch, no we want to make sure caching does not occur under any circumstances - I'll update with a test